src/applications/accreditation/21a/config/submit-transformer.js

```diff
@@ -1,9 +1,271 @@
-import { transformForSubmit as platformTransformForSubmit } from '~/platform/forms-system/exportsFile';
+import {
+  ADDRESS_TYPE_ID,
+  EMPLOYMENT_STATUS_ID,
+  SERVICE_BRANCH_ID,
+  DISCHARGE_TYPE_ID,
+  PHONE_TYPE_ID,
+  DEGREE_TYPE_ID,
+  ADMITTANCE_TYPE_ID,
+  RELATION_TO_APPLICANT_ID,
+} from './enums';
 
-const transformForSubmit = (formConfig, form) => {
-  return platformTransformForSubmit(formConfig, form, {
-    allowPartialAddress: true,
-  });
+const transformForSubmit = (_formConfig, form) => {
+  const transformedData = {};
+
+  const {
+    fullName,
+    placeOfBirth,
+    homeAddress,
+    workAddress,
+    militaryServiceExperiences,
+    employers,
+    educationalInstitutions,
+    jurisdictions,
+    agenciesOrCourts,
+    characterReferences,
+  } = form.data;
+
+  const setName = () => {
+    transformedData.firstName = fullName.first;
+    transformedData.lastName = fullName.last;
+
+    if (fullName?.middle) transformedData.middleName = fullName.middle;
+    if (fullName?.suffix) transformedData.suffix = fullName.suffix;
+  };
+
+  const setBirth = () => {
+    transformedData.birthdate = form.data.dateOfBirth;
+    transformedData.birthCity = placeOfBirth.city;
+    transformedData.birthState = placeOfBirth.state;
+    transformedData.birthCountry = placeOfBirth.country;
+  };
+
+  const setHomeAddress = () => {
+    transformedData.homeAddress = {
+      addressType: homeAddress.isMilitary,
+      line1: homeAddress.street,
+      city: homeAddress.city,
+      postalCode: homeAddress.postalCode,
+      country: homeAddress.country,
+    };
+
+    transformedData.homeAddressId = ADDRESS_TYPE_ID.home;
+
+    if (homeAddress?.street2)
+      transformedData.homeAddress.line2 = homeAddress.street2;
+  };
+
+  const setContactInfo = () => {
+    transformedData.homePhone = form.data.phone;
+    transformedData.phoneTypeId = PHONE_TYPE_ID[form.data.typeOfPhone];
+    transformedData.phoneType = {
+      name: form.data.typeOfPhone,
+    };
+    transformedData.homeEmail = form.data.email;
+  };
+
+  const setEmployment = () => {
+    const employmentStatus = Object.keys(form.data.employmentStatus)[0];
+    transformedData.employmentStatus = employmentStatus;
+    transformedData.employmentStatusId = EMPLOYMENT_STATUS_ID[employmentStatus];
+    if (form.describeEmployment) {
+      transformedData.employmentStatusExplanation = form.describeEmployment;
+    }
+
+    if (employmentStatus === 'employed') {
+      transformedData.businessAddress = {
+        addressType: workAddress.isMilitary,
+        line1: workAddress.street,
+        city: workAddress.city,
+        state: workAddress.state,
+        postalCode: workAddress.postalCode,
+        country: workAddress.country,
+      };
+
+      transformedData.businessAddressId = ADDRESS_TYPE_ID.business;
+
+      if (workAddress.street2)
+        transformedData.businessAddress.line2 = workAddress.street2;
+    }
+  };
+
+  const setMilitaryService = () => {
```

Comment:
The `setMilitaryService` function could be refactored to leverage JS array methods for more concise, readable code. It should also be tested.

Here is an example test file:
https://github.com/department-of-veterans-affairs/vets-website/blob/6d4b0c107a1ddc9f1db02a38bb22fc31b5a0b3ba/src/applications/edu-benefits/0994/tests/submit-transformer.unit.spec.js

Pseudocode of how you could test this (you'll likely need to refactor):
```
describe('transformMilitaryService', () => {
  it('should correctly transform military service experiences', () => {
    const militaryServiceExperiences = [
      {
        explanationOfDischarge: 'Completed service honorably',
        dateRange: { from: '2001-01-01', to: '2005-01-01' },
        branch: 'Army',
        characterOfDischarge: 'Honorable',
      },
      {
        explanationOfDischarge: 'Medical discharge',
        dateRange: { from: '2006-06-01', to: '2008-08-01' },
        branch: 'Navy',
        characterOfDischarge: 'Medical',
      },
    ];

    const expectedOutput = [
      {
        dischargeTypeExplanation: 'Completed service honorably',
        entryDate: '2001-01-01',
        dischargeDate: '2005-01-01',
        serviceBranch: { name: 'Army' },
        dischargeType: { name: 'Honorable' },
        dischargeTypeId: DISCHARGE_TYPE_ID['Honorable'],
        serviceBranchId: SERVICE_BRANCH_ID['Army'],
      },
      {
        dischargeTypeExplanation: 'Medical discharge',
        entryDate: '2006-06-01',
        dischargeDate: '2008-08-01',
        serviceBranch: { name: 'Navy' },
        dischargeType: { name: 'Medical' },
        dischargeTypeId: DISCHARGE_TYPE_ID['Medical'],
        serviceBranchId: SERVICE_BRANCH_ID['Navy'],
      },
    ];

    const transformedData = transformMilitaryService(militaryServiceExperiences);

    expect(transformedData).toEqual(expectedOutput);
  });
});
```
Pseudocode on how you could leverage array methods instead of `for (let i = 0; i < militaryServiceExperiences.length; i += 1) {`
```
const setMilitaryService = () => {
  transformedData.militaryServices = militaryServiceExperiences.map(serviceExp => ({
    dischargeTypeExplanation: serviceExp.explanationOfDischarge,
    entryDate: serviceExp.dateRange?.from,
    dischargeDate: serviceExp.dateRange?.to,
    serviceBranch: { name: serviceExp.branch },
    dischargeType: { name: serviceExp.characterOfDischarge },
    dischargeTypeId: DISCHARGE_TYPE_ID[serviceExp.characterOfDischarge],
    serviceBranchId: SERVICE_BRANCH_ID[serviceExp.branch],
  }));
};
```